conans/client/installer.py

```diff
@@ -397,15 +397,8 @@ def _download(n):
                 _download(node)
 
     def _download_pkg(self, layout, pref, node):
-        conanfile = node.conanfile
-        package_folder = layout.package(pref)
-        output = conanfile.output
-        with set_dirty_context_manager(package_folder):
```

Comment:
The ``dirty_context_manager`` was only applied for ``conan install`` command, but not for ``conan download`` commands, which could also leave the packages broken if interrupted. This has been moved to ``RemoteManager``, so no matter who calls the download of a package, the ``.dirty`` mechanism will be used.